DeeployTest/testUtils/dmaUtils.py

```diff
@@ -0,0 +1,362 @@
+        memoryMap[memoryStart][-1].append(MemoryBlock(tensorName, memoryStart, lifetime, addrSpace))
+
+    addrSpacePing = (0, inputSizeInBytes)
+    addrSpacePong = (inputSizeInBytes, 2 * inputSizeInBytes)
+
+    ## First input tensor has a special lifetime (0, 0)
+    appendMemoryMapStart(graph.nodes[0].inputs[0].name, (0, 0), addrSpacePing)
+
+    for i, node in enumerate(graph.nodes):
+        # Start with addrSpacePong because we used "Ping" for the first input tensor
+        appendMemoryMapStart(node.outputs[0].name, (i, i + 1), addrSpacePong if i % 2 == 0 else addrSpacePing)
+
+    ## Set the rest
+
+    def setMemoryMapRest(memory: str, inputAddrSpace: Tuple[int, int], outputAddrSpace: Tuple[int, int]) -> None:
+        for i, node in enumerate(graph.nodes):
+            # Empirically concluded from looking at produced memory maps
+            if i + 1 == len(graph.nodes):
+                endLifetime = i + 1
+            else:
+                endLifetime = i
+
+            memoryMap[memory][i].extend([
+                MemoryBlock(name = node.inputs[0].name, level = memory, lifetime = (i, i), addrSpace = inputAddrSpace),
+                MemoryBlock(name = node.outputs[0].name,
+                            level = memory,
+                            lifetime = (i, endLifetime),
+                            addrSpace = outputAddrSpace),
+            ])
+
+    if memoryMultibuffer is not None:
+        setMemoryMapRest(memoryMultibuffer, inputMultibufferAddrSpace, outputMultibufferAddrSpace)
+
+    for memory in singleTileMemories:
+        setMemoryMapRest(memory, inputTileAddrSpace, outputTileAddrSpace)
+
+    return tilingSolution, memoryMap
+
+
+def defaultScheduler(graph: gs.Graph) -> List[List[gs.Node]]:
+    return [[node] for node in graph.nodes]
+
+
+def setup_pulp_deployer(defaultMemory: str, targetMemory: str, graph: gs.Graph, inputTypes: Dict[str, Type[Pointer]],
+                        doublebuffer: bool) -> NetworkDeployer:
+    L3 = MemoryLevel(name = "L3", neighbourNames = ["L2"], size = 64000000)
+    L2 = MemoryLevel(name = "L2", neighbourNames = ["L3", "L1"], size = 1024000)
+    L1 = MemoryLevel(name = "L1", neighbourNames = ["L2"], size = 64000)
+    memoryLevels = [L3, L2, L1]
+    memoryLevelMap = {mem.name: mem for mem in memoryLevels}
+
+    assert defaultMemory in memoryLevelMap, f"defaultMemory {defaultMemory} is not part of PULP's memory hierarchy {list(memoryLevelMap.keys())}"
+    assert targetMemory in memoryLevelMap, f"targetMemory {targetMemory} is not part of PULP's memory hierarchy {list(memoryLevelMap.keys())}"
+
+    memoryHierarchy = MemoryHierarchy(memoryLevels)
+    memoryHierarchy.setDefaultMemoryLevel(defaultMemory)
+
+    platform = MemoryPULPPlatform(memoryHierarchy, memoryLevelMap[targetMemory])
+
+    deployer = PULPDeployer(graph, platform, inputTypes, PULPOptimizer, defaultScheduler, default_channels_first = True)
```

Comment:
Please make sure that the `deeployStateDir` is located in the correct position. You can use this patch:

[Patch.patch](https://github.com/user-attachments/files/21969581/Patch.patch)

(Comment also applies to the other deployer)

Reply:
Applied in 751e89b1812a